Engine/Main/GameObject.cpp

```diff
@@ -547,7 +574,35 @@ void GameObject::CopyComponents(const GameObject& gameobject_to_copy)
 			this->components.push_back(copy);
 		}
 	}
+	RemoveComponentsCopying(gameobject_to_copy);
+}
 
+void GameObject::CopyComponents(const GameObject& gameobject_to_copy)
+{
+	this->components.reserve(gameobject_to_copy.components.size());
+	for (const auto& component : gameobject_to_copy.components)
+	{
+		component->modified_by_user = false;
+		//Component * my_component = GetComponent(component->type); //TODO: This doesn't allow multiple components of the same type
+		Component* copy = nullptr;
+		if(component->type != Component::ComponentType::SCRIPT)
+		{
+			copy = component->Clone(this->original_prefab);
```

Comment:
clone is not working with the component scripts?

Reply:
With component script the clone method you implemented what it does is to copy literally the component that u are calling, and that wont work as scripts have inside its owner that in this case is different

Reply:
Not only that, the problem is that you can not copy the same instance script have, because then both objects will have the same behaviour at the same time (ex: if I move one, the other will move also) so the script instance should be unique and independent. So in order to accompilsh that you need to create another instance of the same script. (Hope you understand that, I explain myself quite bad)